### ISSUE TITLE
Vaadin 23.3.2: Building for production fails "Unable to resolve reference $@vaadin/vaadin-material-styles"

### ISSUE THREAD

**Issue by AB-xdev** [vaadin/flow#15662]
### Description of the bug

When running a production build (in the latest version of Vaadin 23.3.2) and having no ``package.json``/``package-lock.json`` it fails with:
```
...
[WARNING] package-lock.json is missing from this project. This may cause the npm package installation to take several minutes. It is recommended to keep the package-lock.json file persistently in your project. Please stand by...
npm ERR! Unable to resolve reference $@vaadin/vaadin-material-styles

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\...\npm-cache\_logs\2023-01-10T13_37_48_524Z-debug-0.log
[ERROR] Command `C:\Program Files\nodejs\npm.cmd --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install` failed:

[ERROR] >>> Dependency ERROR. Check that all required dependencies are deployed in npm repositories.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:45 min
[INFO] Finished at: 2023-01-10T14:39:09+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:23.3.2:build-frontend (default) on project mytodo: Could not execute build-frontend goal: Npm install has exited with non zero status. Some dependencies are not installed. Check npm command output -> [Help 1]
```

### Expected behavior

No error occurs

### Minimal reproducible example

* Download the quick starter: https://vaadin.com/docs/latest/guide/quick-start (for future users ensure that the version is 23.3.2)
* Delete the ``package.json`` and ``package-lock.json``
* Run the build for production ``mvnw clean package -Pproduction``
* The problem occurs

### Versions

- Vaadin / Flow version: 23.3.2
- Java version: 17
- OS version: Doesn't matter; tested with Win 10 and Linux/Docker

Installed NodeJS-Version: v18.13.0 (latest LTS)
Installed NPM-Version: 9.2.0 (latest)

**Comment by AB-xdev** [comment]
Update:
So apparently the latest version is [``23.3.3``](https://github.com/vaadin/platform/releases/tag/23.3.3) now which was released a few hours ago (guess https://vaadin.com/releases wasn't updated yet) and the problem seems to be fixed there.

However I couldn't find any issue/PR that fixes the above problem inside the release notes...

**Comment by Artur-** [comment]
It sounds very similar to the npm issue described in https://github.com/npm/cli/issues/6028

Created https://github.com/vaadin/flow/issues/15667 so that we can avoid these issues until npm 9 is fixed

**Comment by Artur-** [comment]
This was fixed in npm 9.3.0
